packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/BillingManager.java

```diff
@@ -144,13 +152,26 @@ public void run() {
           return;
         }
 
-        BillingFlowParams.Builder purchaseParams = BillingFlowParams.newBuilder()
-          .setSkuDetails(skuDetails);
+        BillingFlowParams.Builder purchaseParams = BillingFlowParams.newBuilder().setSkuDetails(skuDetails);
+
         if (oldPurchaseToken != null) {
           purchaseParams.setSubscriptionUpdateParams(
             BillingFlowParams.SubscriptionUpdateParams.newBuilder().setOldSkuPurchaseToken(oldPurchaseToken).build()
           );
         }
+
+        /**
+         * For Android billing to work without a 'Something went wrong on our end. Please try again.'
+         * error, we must provide BOTH obfuscatedAccountId and obfuscatedProfileId.
+         */
+        if (obfuscatedAccountId != null && obfuscatedProfileId != null) {
+          purchaseParams.setObfuscatedAccountId(obfuscatedAccountId);
+          purchaseParams.setObfuscatedProfileId(obfuscatedProfileId);
```

Comment:
Perhaps I'm missing something (I'm just doing a static analysis review) but if this is to set both (that is, Google expects both values), then this should be an OR instead of AND? That, or the docblock needs updating?

Reply:
@bradjones1 not sure I follow, we need to provide obfuscatedAccountId **AND** obfuscatedProfileId, if we don't then the payment will just fail with 'Something went wrong on our end. Please try again.'. Not sure why this is the case, nor could I find anything on Google Documentation about this requirement, but from testing it was the case.

Reply:
@lukebrandonfarrell Correct... As I read this `if`, though, it could be summarized as, "If we have *both* accountId and profileId, then set them both. This requires the caller to then set both, correct?

I'm concerned that, given how unintuitive this is, that callers might set one but not the other (likely, the account ID and not the profile) but this would end in a no-op in this code. Is it possible we can throw a warning when both aren't provided, or perhaps if account is provided but not profile, we set a default for the other value?

Reply:
The requirement to set both fields is in the docs and the TS types. If Android changes the API we can change this module in the future.

Reply:
So this is mostly to enforce the requirement in non-TS environments? Seems to me then that we could/should throw a runtime assertion type error? But I will admit I am just now wading into this kind of question in the codebase.